samcli/lib/deploy/deployer.py

```diff
@@ -424,6 +431,9 @@ def describe_stack_events(self, stack_name, time_stamp_marker, **kwargs):
                 # Reset retry attempts if iteration is a success to use client_sleep again
                 retry_attempts = 0
             except botocore.exceptions.ClientError as ex:
+                if "Stack with id {0} does not exist".format(stack_name) in str(ex) or on_failure == FailureMode.DELETE:
```

Comment:
let's change the or to and as discussed in code review session, as we don't want to accidentally catch the "Stack with id {0} does not exist" exception when the on failure is not specified
```suggestion
                if "Stack with id {0} does not exist".format(stack_name) in str(ex) and on_failure == FailureMode.DELETE:
```